### PR TITLE
Week 5 corey arnold

### PR DESCRIPTION
**Pull request by arnoldc94**
# Submitting Week 5 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [ ✅ ] JavaScript Coding Challenges
- [ ✅ ] Ruby Coding Challenges
- [ ✅ ] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:
I would like to know my ranking if possible.

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -15,42 +15,165 @@
...
+    }else if(value.toLowerCase() === 'e'){
+        return 3
+    }else if(value.toLowerCase() === 'i'){
+        return 1
+    }else if(value.toLowerCase() === 'o'){
+        return 0
+    }else{
+        return value
+    }
+}).join("")
```
Comment (elysemontano) [inline code-challenges.test.js:64]: 💯

File: code-challenges.test.js [hunk 2]
```diff
@@ -15,42 +15,165 @@
...
+// Create an if conditional statement that compares the typeof the letter paramater strictly equal to the 'string' datatype
+// return `${letter} is not a letter` if true
+// else filter the array paramater to iterate over the array
+// On each iteration set the value.toLowerCase() and check if it .includes() the letter paramater letter.toLowerCase()
+// This will return the strings that include the passed in letter
+// Outside of jest testing console.log the function and paramaters to see the result in the terminal
+
+const matchLetter = (arr,letter) =>{
+     if(typeof letter !== 'string'){
+        return `${letter} is not a letter`
```
Comment (elysemontano) [inline code-challenges.test.js:110]: Nice addition to consider edge cases!

File: code-challenges.test.js [hunk 3]
```diff
@@ -15,42 +15,165 @@
...
+    if(newArr.length === 3){
+        if((newArr[0]===newArr[1]&&newArr[0]!==newArr[2]) || (newArr[0]===newArr[2]&&newArr[0]!==newArr[1])){
+            return true
+        }else{
+            return false
+        }
+    }else{
+        return false
+    }
+}
```
Comment (elysemontano) [inline code-challenges.test.js:170]: Something to consider, what would happen if we tested the array [5, 5, 3, 3, 3]?  This should return true, but it looks like it may currently return false with your function.

File: interview-questions.md [hunk 4]
```diff
@@ -6,46 +6,47 @@ First, without external resources. Challenge yourself to answer from memory.
...
-Researched answer:
+Researched answer: A relational database is a collection of data with relationships between the data. The data is stored in tables made up of column and rows. These are used to hold data about objects that are represented in the database. There are other types of databases, which are, hierarchical, network, object-oriented.
 
 5. What are primary keys? Why are they important?
 
-Your answer:
+Your answer: A primary key is a unique value that is given to data stored in a database or even instances of classes in ruby. The primary key is used to access that specific data to update, read it or delete it. It must be given to the data when created. 
 
-Researched answer:
+Researched answer: A primary key is a column or multiple columns that act as a unique identifier for each row of data in a table. A primary key is neccissary to create, read, update or delete a piece of data.
```
Comment (elysemontano) [inline interview-questions.md:38]: Great work in this section!

File: ruby-challenges.rb [hunk 5]
```diff
@@ -33,3 +57,55 @@
...
+# pass the model as a paramater in the initilize method.
+# create a bike_info method that returns "The #{model} bike has #{wheels} wheels and is going #{speed} mph."
+# create a pedal_faster method that passes in a num paramater
+# return the speed equals speed plus the num paramater.
+# create a brake method that passes in a num paramater.
+# set speed to the number being passed minus the speed.
+# create an if conditional statement that compares if speed is less than zero then return the speed is equal to zero.
+# else the speed is the current positive number
+# create a new instance of the class named my_bike and pass the model as 'Trek'.
+# print the my_bike instance invoking each new method and pass in a num paramater as needed.
```
Comment (elysemontano) [inline ruby-challenges.rb:75]: Really loving the pseudocode for the Ruby section!

**Review by elysemontano — APPROVED** [review]
Excellent work as always Corey!  Your skill acquisition is at a competent level as you continue to show a thorough understanding of the concepts.  I really like the fact that you are thinking about edge cases in your code.  As you keep progressing and if you are looking to stretch yourself even further, I want to encourage you to include tests to evaluate those edge cases as well.  Great job!
